valhalla/mjolnir/osmaccessrestriction.h

```diff
@@ -59,7 +59,7 @@ class OSMAccessRestriction {
   uint64_t value_;
 
   struct Attributes {
-    uint16_t type_ : 4;
+    uint16_t type_ : 6; // interpreted as AccessType but stored in AccessRestriction::type_
     uint16_t modes_ : 12;
```

Comment:
ok so i dont think we should merge this and this is why. please note that we are storing this data as 16 bit value. if you increase `type_` to cover 6 bits now we have 18 bits. the compiler will pad this struct to 32 bits but its not what the code says. a gotcha for people to stumble over in the future.

we have two options:

1. revert `type_` back to 4 bits because really we dont ever use the other 2 bits yet and then add a compile time check that max AccessType is < 2^4
2. upgrade attributes to 32bits: `struct Attributes { uint32_t type_ : 6; uint32_t modes_ : 12; uint32_t spare_ : 14; };`

i favor the first option because doubling the size of this struct is wasteful (though its relatively sparsely populated so not so bad)

Reply:
Thanks for clarifying me

Reply:
@dnesbitt61 also mentioned that we could just go into `AccessRestriction::type_` set the type to 4 bits and then have 2 bits spare in that field